src/nfa/nfa.cc

```diff
@@ -165,45 +163,7 @@ namespace {
         TarjanNodeData() = default;
 
         TarjanNodeData(State q, const Delta & delta, unsigned long index)
-            : post_it(delta[q].cbegin()), post_end(delta[q].cend()), index(index), lowlink(index), initilized(true), on_stack(true) {
-            if (post_it != post_end) {
-                targets_it = post_it->cbegin();
-                targets_end = post_it->cend();
-            }
-        };
-
-        // TODO: this sucks. In fact, if you want to check that you have the last sucessor, you need to
-        // first align the iterators.
-        // TODO: this is super-ugly. If we introduce Post::transitions iterator, this could be much easier.
-        // Align iterators in a way that the current state is stored in *(this->targets_it).
-        void align_succ() {
-            while (this->post_it != this->post_end && this->targets_it == this->targets_end) {
-                if (this->targets_it == this->targets_end) {
-                    ++this->post_it;
-                    if (this->post_it != this->post_end) {
-                        this->targets_it = this->post_it->cbegin();
-                        this->targets_end = this->post_it->cend();
-                    }
-                }
-            }
-        }
-
-        State get_curr_succ() {
-            align_succ();
-            return *(this->targets_it);
-        }
-
-        void move_to_next_succ() {
-            if(this->post_it == this->post_end) {
-                return;
-            }
-            ++this->targets_it;
-        }
-
-        bool is_end_succ() {
-            align_succ();
-            return this->post_it == this->post_end;
-        }
+            : current_move(delta[q].moves().begin()), end_move(delta[q].moves().end()), index(index), lowlink(index), initilized(true), on_stack(true) {};
```

Comment:
I advise extracting the call to `moves()` to the constructor body into a local variable and initialize `current_move` and `end_move` there in order to prevent creating an instance of `Moves` twice.

Reply:
Done.